base/src/main/java/org/aya/concrete/stmt/TopLevelDecl.java

```diff
@@ -2,20 +2,29 @@
 // Use of this source code is governed by the MIT license that can be found in the LICENSE.md file.
 package org.aya.concrete.stmt;
 
+import org.aya.concrete.Expr;
 import org.aya.core.def.GenericDef;
+import org.aya.resolve.context.Context;
 import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
 
 /**
  * Generic concrete definition, corresponding to {@link GenericDef}.
  *
  * @author zaoqi
  */
-public sealed interface TopLevelDecl extends GenericDecl permits ClassDecl, Decl {
+public sealed interface TopLevelDecl extends GenericDecl, Stmt permits ClassDecl, Decl {
```

Comment:
What about making it an abstract class, and make the getters/setters fields?

Reply:
I had considered it, but there is a class called `Signatured` on the way

Reply:
:(